src/models/transaction.ts

```diff
@@ -108,7 +108,7 @@ class Transaction {
     arr.push(helpers.intToBytes(this.version, 2))
 
     // Len tokens
-    if (this.tokens.length) {
+    if (this.version !== CREATE_TOKEN_TX_VERSION) {
```

Comment:
I feel we shouldn't mix both cases in a single serialization code. Isn't it better to have the following methods: `blockSerialization()`, `txSerialization()`, `createTokenTxSerialization()`? They could call helpers to prevent code duplication.

Reply:
Block and transactions can be handled the same way but I agree we have some custom serializations for the create token tx. Should I create a class `CreateTokenTransaction` that extends this `Transaction` class instead of having three different methods in the same class?

Reply:
I guess so. Even blocks and transactions might differ significantly if we serialize MergeMinedBlocks for instance.

Reply:
Done here: https://github.com/HathorNetwork/hathor-wallet-lib/pull/184/commits/552aa00899aab7cc45b80bfac8254bcdbac7c335

Reply:
Great! Thanks!